lib/Doctrine/MongoDB/Cursor.php

```diff
@@ -151,16 +151,14 @@ public function count($foundOnly = false)
      * @see http://php.net/manual/en/mongocursor.current.php
      * @return array|null
      */
-    public function current()
-    {
-        $current = $this->mongoCursor->current();
-        if ($current instanceof \MongoGridFSFile) {
-            $document = $current->file;
-            $document['file'] = new GridFSFile($current);
```

Comment:
See https://github.com/doctrine/mongodb/issues/252#issuecomment-198002727. I think it may be wiser to conditionally skip this injection based on the query's projection.